python/cuml/experimental/explainer/permutation_shap.pyx

```diff
@@ -0,0 +1,420 @@
+#
+# Copyright (c) 2020, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import cuml
+import cupy as cp
+import numpy as np
+
+from cudf import DataFrame as cu_df
+from cuml.common.array import CumlArray
+from cuml.common.input_utils import input_to_cuml_array
+from cuml.common.logger import warn
+from cuml.common.logger import info
+from numba import cuda
+from pandas import DataFrame as pd_df
+
+from cuml.raft.common.handle cimport handle_t
+from libcpp cimport bool
+from libc.stdint cimport uintptr_t
+
+
+try:
+    from shap.utils import partition_tree_shuffle
+    shap_not_found = False
```

Comment:
Maybe check that we have a usable (new) version of shap and warn and fail otherwise?